internal/raft/append_entries_test.go

```diff
@@ -0,0 +1,77 @@
+package raft
+
+import (
+	"net"
+	"testing"
+
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/assert"
+	"github.com/tomarrell/lbadd/internal/network"
+	"github.com/tomarrell/lbadd/internal/raft/cluster"
+	"github.com/tomarrell/lbadd/internal/raft/message"
+)
+
+func TestAppendEntries(t *testing.T) {
```

Comment:
What exactly does this test do? At the bottom of this function, there is a huge block of assertions, but generally, the code in this function is not documented, and I can't retrace what exactly this test does and tests.

Reply:
Yes please, write what the test does as documentation.